### ISSUE TITLE
Building with CNTKCustomMKL:  fatal error LNK1181: cannot open input file 'CNTKLibrary-2.0.lib'

### ISSUE THREAD

**Issue by quicktrick** [Microsoft/CNTK#598]
Everything is set up according to https://github.com/Microsoft/CNTK/wiki/Setup-CNTK-on-Windows

```
9>------ Build started: Project: V2LibraryTests, Configuration: Release x64 ------
.....
9>LINK : fatal error LNK1181: cannot open input file 'CNTKLibrary-2.0.lib'
.....
```
```
Error	1	error LNK1181: cannot open input file 'CNTKLibrary-2.0.lib'	d:\...\CNTK\Tests\UnitTests\V2LibraryTests\LINK	V2LibraryTests
```

**Comment by mahilleb-msft** [comment]
Project dependencies of V2LibraryTests are not right (this might not be MKL specific). I will look into this.
In the mean time, can you try if a rebuild fixes your build (that also would indicate project dependency issues)?

Thanks, Mark

**Comment by quicktrick** [comment]
You are right Mark, the rebuild has fixed the build.  Thank you!

Alex

**Comment by FilipTolj** [comment]
I have the same issue LNK1181: cannot open input file 'CNTKLibrary-2.0.lib'. The projects which are affected  "V2LibraryTests" "NetworkTests" "V2LibraryDistributionTests" "CNTK". I've rebuilt it several times still the same issue. Hope somebody can help.

**Comment by navidnadery** [comment]
I have the same problem. The rebuild has not fixed the build. Even I installed NuGet packages as described in https://www.nuget.org/packages/Microsoft.Research.CNTK.CpuEval-mkl/2.0.0-rc3.
Here is my LINK error and its previous line in VS output: 

CNTKLibraryManagedDll -> C:\repos\cntk\x64\Debug_CpuOnly\Cntk.Core.Managed-2.0rc3d.dll
LINK : fatal error LNK1104: cannot open file 'Cntk.Core-2.0rc3d.lib'

**Comment by zhouwangzw** [comment]
@navidnadery Were you building the cntk.sln or building [Eval examples](https://github.com/Microsoft/CNTK/tree/master/Examples/Evaluation/CNTKLibraryCSEvalCPUOnlyExamples)? If you want to build cntl.sln, you do not need to install Nuget packages. If you want build eval examples, you need to install CNTK.CPUOnly or CNTK.GPU (for GPU usage) nuget packages. Please find details about Evaluation [here](https://docs.microsoft.com/en-us/cognitive-toolkit/CNTK-Evaluation-Overview).

**Comment by navidnadery** [comment]
@zhouwangzw thank you for your good explanation. I am building the cntk.sln. I found the problem, I installed VS Update 3 apparently after installing Visual Studio 2015 + 'Common Tools for Visual C++ 2015' checked, but it seems that it has a bug because installing Update 3 removes 'Common Tools for Visual C++ 2015' and it should be installed again. It worked for me. I tried twice and in both update 3 removed 'Common Tools for Visual C++ 2015' check.
